internal/pipe/snapcraft/snapcraft.go

```diff
@@ -373,9 +384,31 @@ func link(src, dest string, mode os.FileMode) error {
 		if info.IsDir() {
 			return os.MkdirAll(dst, info.Mode())
 		}
-		if err := os.Link(path, dst); err != nil {
-			return err
+		if err := copyFile(path, dst, mode); err != nil {
+			return fmt.Errorf("fail copy file '%s': %w", path, err)
 		}
-		return os.Chmod(dst, mode)
+		return nil
 	})
 }
+
+func copyFile(src, dst string, mode os.FileMode) error {
```

Comment:
curious why this change? did you find any issues with the hard linking one?

Reply:
Yes, hard link work only if both files are on the same FS.
My `/tmp` is managed outside, so tests are failing with:
```
failed to link completer: link testdata/foo-completer.bash /tmp/TestCompleter1448278326/001/dist/foo_arm6/prime/testdata/foo-completer.bash: invalid cross-device link
```
Generally, hardlinks are a bad idea.
I tried to use Symlink but `chmod` is inherited and failed.